extensions-core/druid-basic-security/src/main/java/io/druid/security/basic/db/derby/DerbySQLBasicSecurityStorageConnector.java

```diff
@@ -0,0 +1,191 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.security.basic.db.derby;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Supplier;
+import com.google.common.collect.ImmutableList;
+import com.google.inject.Inject;
+import io.druid.guice.ManageLifecycle;
+import io.druid.java.util.common.StringUtils;
+import io.druid.java.util.common.lifecycle.LifecycleStart;
+import io.druid.java.util.common.logger.Logger;
+import io.druid.metadata.MetadataStorage;
+import io.druid.metadata.MetadataStorageConnectorConfig;
+import io.druid.metadata.MetadataStorageTablesConfig;
+import io.druid.security.basic.BasicAuthConfig;
+import io.druid.security.basic.db.SQLBasicSecurityStorageConnector;
+import org.apache.commons.dbcp2.BasicDataSource;
+import org.skife.jdbi.v2.DBI;
+import org.skife.jdbi.v2.Handle;
+
+@ManageLifecycle
+public class DerbySQLBasicSecurityStorageConnector extends SQLBasicSecurityStorageConnector
+{
+  private static final Logger log = new Logger(DerbySQLBasicSecurityStorageConnector.class);
+
+  private final DBI dbi;
+  private final MetadataStorage storage;
+  private static final String QUOTE_STRING = "\\\"";
+
+  @Inject
+  public DerbySQLBasicSecurityStorageConnector(
+      MetadataStorage storage,
+      Supplier<MetadataStorageConnectorConfig> config,
+      Supplier<BasicAuthConfig> basicAuthConfigSupplier,
+      ObjectMapper jsonMapper
+  )
+  {
+    super(config, basicAuthConfigSupplier, jsonMapper);
+
+    final BasicDataSource datasource = getDatasource();
+    datasource.setDriverClassLoader(getClass().getClassLoader());
+    datasource.setDriverClassName("org.apache.derby.jdbc.ClientDriver");
+
+    this.dbi = new DBI(datasource);
+    this.storage = storage;
+    log.info("Derby connector instantiated with metadata storage [%s].", this.storage.getClass().getName());
+  }
+
+  public DerbySQLBasicSecurityStorageConnector(
```

Comment:
Looks not being used.

Reply:
It's not instantiated directly, but the class is bound in BasicSecurityDruidModule:

```
    PolyBind.optionBinder(binder, Key.get(BasicSecurityStorageConnector.class))
            .addBinding("derby")
            .to(DerbySQLBasicSecurityStorageConnector.class)
            .in(ManageLifecycle.class);

    PolyBind.optionBinder(binder, Key.get(SQLBasicSecurityStorageConnector.class))
            .addBinding("derby")
            .to(DerbySQLBasicSecurityStorageConnector.class)
            .in(ManageLifecycle.class);
```

Reply:
Ah ok.